source/features/scroll-to-current-file.tsx

```diff
@@ -0,0 +1,57 @@
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+import GitHubFileURL from '../github-helpers/github-file-url.js';
+
+function scrollToCurrentFile(): void {
+	const url = new GitHubFileURL(location.href);
+	const filePath = url.filePath;
+
+	// we need to escape the filePath if we use querySelector.
+	// eslint-disable-next-line unicorn/prefer-query-selector
+	const item = document.getElementById(`${filePath}-item`);
+	const text = item?.querySelector('.PRIVATE_TreeView-item-content-text');
+	text?.scrollIntoView({block: 'center', behavior: 'auto'});
+}
+
+function init(signal: AbortSignal): void {
+	signal.addEventListener('abort', unload);
+
+	// After the file navigation
+	window.addEventListener('turbo:load', scrollToCurrentFile);
+
+	// After expanding the file tree
+	document.addEventListener('soft-nav:end', scrollToCurrentFile);
+
+	// After navigating back to a repo tree or a single file page from other tabs (e.g. Issues, Pull requests)
+	scrollToCurrentFile();
+}
+
+function unload(): void {
+	// After the file navigation
+	window.removeEventListener('turbo:load', scrollToCurrentFile);
+
+	// After expanding the file tree
+	document.removeEventListener('soft-nav:end', scrollToCurrentFile);
+}
+
+void features.add(import.meta.url, {
+	asLongAs: [
+		pageDetect.isLoggedIn,
+	],
+	include: [
+		pageDetect.isRepoTree,
+		pageDetect.isSingleFile,
+	],
+	init,
+	awaitDomReady: true,
+});
```

Comment:
I think you only need this according to what I mentioned in my comment

```suggestion
function init(): void {
	scrollToCurrentFile();
}

void features.add(import.meta.url, {
	include: [
		pageDetect.isRepoTree,
		pageDetect.isSingleFile,
	],
	exclude: [
		pageDetect.isRepoRoot,
	],
	awaitDomReady: true,
	init,
});
```

Reply:
It seems that GitHub adjusts the layout/scroll multiple times after the DOM is ready. It jiggles on the load when `turbo:load` listener is removed.

https://github.com/user-attachments/assets/49efc45e-e450-4bc4-88b3-f41e6d7ed6bb

It seems that `soft-nav:end` is not needed anymore since the file tree works fine by default. (It used to be problematic)

Reply:
`turbo:load` is fired at every page load so we don't want to mess with the native scrolling. You might want to find another event or just use `setTimeout(scrollToCurrentFile, 500)` inside `init`

Reply:
"turbo:load" has a similar lifecycle to `init` func.
It isn't fired when you just navigate between files.
It is fired only when you navigate between file/dir <-> Root, Issues, Pull Requests, etc.

Also, I found that the invocation timing of `init` is inconsistent.
It is called before when you open an new link, but it is called after `turbo:load` when you navigate back to the file/directory view.
It might be due to the preceding `async/await` in the `setupPageLoad`.

Reply:
I could fix the event timing, but it seems messy to play with `Promise`. I'll just stick to `setTimeout`